js/main.js

```diff
@@ -1,108 +1,123 @@
 }
 
-function main(){
-  // Eww JSONP (Thanks, CORS!)
-  var lunchtag = document.createElement("script");
-  lunchtag.src = "https://melroseschools.nutrislice.com/menu/api/weeks/school/melrose/menu-type/lunch/" + today.getFullYear() + "/00/00/?format=json-p&callback=getLunchInfo";
-  document.getElementsByTagName('head')[0].appendChild(lunchtag);
-
-  // Aspen Stuff
-  loggedOutAspen = new HttpClient();
-  try {
-    loggedOutAspen.get('https://mhs-aspencheck-serve.herokuapp.com', function(response) {
-      var aspenInfo = JSON.parse(response);
-      var lastUpdated = new Date(aspenInfo.asOf*1000);
-      var block = (aspenInfo.schedule.block);
-      var blockSchedule = (aspenInfo.schedule.blockSchedule);
-      var day = (aspenInfo.schedule.day);
-      var blockOfDay = (aspenInfo.schedule.blockOfDay);
-
-      var events = (aspenInfo.calendar.events);
-      var isHalfDay = (aspenInfo.calendar.isHalfDay);
-
-      var percentComplete = (blockOfDay/6)*100;
-
-      clock(isHalfDay);
-
-      document.getElementById('dayNumber').innerHTML = day;
-      document.getElementById('blockId').innerHTML = block;
-      document.getElementById('dayProgress').setAttribute('style', 'width: ' + percentComplete + '%;');
-      document.getElementById('dayProgress').innerHTML = Math.round(percentComplete) + '%';
-      if (block == 'Z') document.getElementById('block-panel').className += " fadeHidden";
-      document.getElementById('lastUpdated').innerHTML = (lastUpdated.getMonth() + 1) + "/" + (lastUpdated.getDate()) + " " + (lastUpdated.getHours()) + ":" + (lastUpdated.getMinutes()) + ":" + (lastUpdated.getSeconds());
-
-      if (events.length > 0){
-        document.getElementById('events-list').innerHTML = "";
-        events.forEach(function(eventTitle){
-          var event = document.createElement('li');
-          event.innerHTML = eventTitle;
-          document.getElementById('events-list').appendChild(event);
+function main() {
+    // Eww JSONP (Thanks, CORS!)
+    var lunchtag = document.createElement("script");
+    lunchtag.src = "https://melroseschools.nutrislice.com/menu/api/weeks/school/melrose/menu-type/lunch/" + today.getFullYear() + "/00/00/?format=json-p&callback=getLunchInfo";
+    document.getElementsByTagName('head')[0].appendChild(lunchtag);
+
+    // Aspen Stuff
+    loggedOutAspen = new HttpClient();
+    try {
+        loggedOutAspen.get('https://mhs-aspencheck-serve.herokuapp.com', function (response) {
+            try {
+                var aspenInfo = JSON.parse(response);
+                var lastUpdated = new Date(aspenInfo.asOf * 1000);
+                var block = (aspenInfo.schedule.block);
+                var blockSchedule = (aspenInfo.schedule.blockSchedule);
+                var day = (aspenInfo.schedule.day);
+
+                var events = (aspenInfo.calendar.events);
+                var isHalfDay = (aspenInfo.calendar.isHalfDay);
+            }catch(err){
```

Comment:
You could also do it as a series of shortened "if exists then this, else set it to default" but too be honest I don't think that's necessary. The two reasons this would fail if you changed the api in such a way that the organization of information is completely different or if the api returns null. If you changed the organization then this would need to be updated anyway so it wouldn't make a difference.